sklearn/utils/validation.py

```diff
@@ -449,6 +449,10 @@ def check_array(array, accept_sparse=False, accept_large_sparse=True,
     if hasattr(array, "dtypes") and hasattr(array.dtypes, '__array__'):
         dtypes_orig = np.array(array.dtypes)
 
+    # extract the underlying numpy ndarray from a Pandas DataFrame
+    if hasattr(array, "iloc"):
```

Comment:
This should not be needed since that we already do an `asarray` below, isn't it?

Reply:
You are right, this conversion is not needed.